docs/architecture/README.md

```diff
@@ -92,7 +92,7 @@ This document specifies the architecture for GCP Ingestion as a whole.
 - Must send messages rejected by transforms to a configurable error destination
    - Must allow error destinations in PubSub and Cloud Storage
 
-### Republisher (Not Yet Implemented)
+### Republisher
```

Comment:
For the Republisher on telemetry data, I'd anticipate the following configuration:

```
--input=projects/moz-fx-data-beam-nonprod-b5f4/subscriptions/telemetry-decoded.republisher

--perDocTypeEnabledList=event
--perDocTypeDestination=projects/moz-fx-data-shar-nonprod-efed/topics/telemetry-decoded-doctype-${document_type}

--perChannelSampleRatios='{"nightly":1.0,"beta":0.1,"release":0.01}'
--perChannelDestination=projects/moz-fx-data-shar-nonprod-efed/topics/telemetry-decoded-sampled-${channel}
```

@sunahsuh requested an event-ping-specific topic and the per-channel topics are as requested in https://github.com/mozilla/gcp-ingestion/issues/396

Which would require creation of a new subscription and of 4 new topics:

```
projects/moz-fx-data-shar-nonprod-efed/topics/telemetry-decoded-doctype-event
projects/moz-fx-data-shar-nonprod-efed/topics/telemetry-decoded-sampled-nightly
projects/moz-fx-data-shar-nonprod-efed/topics/telemetry-decoded-sampled-beta
projects/moz-fx-data-shar-nonprod-efed/topics/telemetry-decoded-sampled-release
```

For the Republisher on structured data, I'd anticipate the following configuration:

```
--input=projects/moz-fx-data-beam-nonprod-b5f4/subscriptions/structured-decoded.republisher
--enableDebugDestination
--debugDestination=projects/moz-fx-data-shar-nonprod-efed/topics/structured-decoded-debug
```

Which would require creation of one new subscription and one new topic. At this point, I don't think it's necessary to break out any docTypes or sample channels for the low volume of structured ingestion data.

cc @whd on the above. Particularly for thoughts on naming.

Reply:
A little late, but after having read https://github.com/mozilla/gcp-ingestion/issues/396 and the above it appears we're going to split the sample out by channel. I don't think we need this level of granularity, and the tradeoff of managing and reading per-channel sampled topics vs. reading the full sampled stream and selecting specific channels as needed seems to me in favor of a producing a single sampled topic. That said, having the sample size vary by channel within a single topic also seems like it could be very wonky from an analysis perspective, depending on how the sample stream is used (for the CEP it would fine). /CC @trink 

For now I'm going to implement this with per-channel topics but I would like to discuss whether this is something we really want.

Reply:
I imagine that the 1% sample of release is going to be a nice go-to for testing or for streaming applications that can suffice on a smaller stream.

That topic is more representative of the overall population than a combined stream of the sample topics would be, so I'm inclined to keep it split up, but I could be convinced otherwise.

Reply:
Ok, if @trink agrees we'll keep it as is. Likely only the CEP will need to read the three streams; the majority of applications will only read the release topic.

Another feature I would like that looks like it may be missing is the ability to generically sample an entire stream, kind of like how the GCS sink is agnostic as to the data it is syncing. With https://github.com/mozilla/gcp-ingestion/issues/482 it looks like we're going to move to sampling by client_id for decoded topics, but I was planning on running a republisher in production on `telemetry-raw` that republishes 1% (random sample as it is now) of the raw topic to stage for the CI dataflow jobs to read and catch issues like https://github.com/mozilla/gcp-ingestion/issues/513.

Reply:
> I was planning on running a republisher in production on telemetry-raw

How about we use "sample_id" if it's present in the message and fall back to a random integer if it's not?

Reply:
I forgot about the channel part. Maybe we just add an additional configuration option for a random sample on the whole stream.

Reply:
That's more or less what I want (since telemetry-raw wont have sample_id), but I would also need the ability to configure "1% of all messages" regardless of channel.